### PR TITLE
Support for collections in where statement

### LINKED ISSUE #99: Support for collections in where statement
```ruby
class Country < ActiveHash::Base
  self.data = [
    {
      id: 1, 
      name: 'Belarus',
      country_code: 'BY'
    },
    {
      id: 3,
      name: 'United States',
      country_code: 'US'
    }
  ]
end
```
Currently it is possible to pass single country code when searching for a country. For example,
```ruby
Country.where country_code: 'BY'
=> #<Country:0x007f94ba2d22a0 @attributes={:id=>1, :name=>"Belarus", :two_letter_code=>"GB"}>
```
It would be great to pass pass an array of country codes when searching for countries, like
```ruby
Country.where country_code: ['BY', 'US']
```

### PR DESCRIPTION
**Pull request by kbrock**
I extended #111 to better support in clauses.

Fixes #99

### CODE REVIEW

**Comment by kbrock** [comment]
@i2bskn is this similar enough to your changes?

**Comment by Hirurg103** [comment]
Yes, @kbrock thank you very much! I love this feature :+1:

**Comment by i2bskn** [comment]
@kbrock Yes, That I wanted to do it has been achieved. Thanks!

**Comment by SmileZero** [comment]
@kbrock :+1: Love this so much.
Just one question. What if the keys in options is a string type( except "id" ). 
I mean the following statement. I know maybe it is not active_hash style.
```
expect(Country.where("name" => %w(US Canada)).map(&:name)).to match_array(%w(US Canada))
```

File: spec/active_hash/base_spec.rb [hunk 1]
```diff
@@ -311,6 +323,19 @@ class Region < ActiveHash::Base
...
       record.name.should == 'US'
     end
+
+    it "finds the record with the specified id as a string" do
+      record = Country.find_by(:id => '1')
+      record.name.should == 'US'
+    end
+
+    it "returns the record that matches options" do
+      expect(Country.find_by(:name => "US").id).to eq(1)
```
Comment (kbrock) [inline spec/active_hash/base_spec.rb]: so the idea is to use `"name"` as the key here and have it work?

Comment (SmileZero) [inline spec/active_hash/base_spec.rb]: Yes. Does it sounds reasonable?

Comment (kbrock) [inline spec/active_hash/base_spec.rb]: looks like `find_by` and `where` and others do not work with string keys. Find by `"id"` does seem to work as a string.

Seems like a separate feature request

Comment (SmileZero) [inline spec/active_hash/base_spec.rb]: :+1: Yeah, I agree that a separate feature request is better.

**Comment by tricknotes** [comment]
This PR is what I want.

@zilkey 
Could you release a new version?

**Pull request merged**